Plugin/QmlApi/source/QmlArcGISArView.cpp

```diff
@@ -0,0 +1,270 @@
+  if (!currentViewpointCamera)
+    return nullptr;
+
+  // converts the point from the screen space to the 3D space
+  const std::array<double, 7> hitResult = ArcGISArViewInterface::hitTestInternal(x, y);
+
+  // calculates the matrix transformations and creates the QmlPoint location.
+  QVariant location;
+  QMetaObject::invokeMethod(m_qmlWrapper, "screenToLocation", Qt::DirectConnection,
+                            Q_RETURN_ARG(QVariant, location),
+                            Q_ARG(QVariant, QVariant::fromValue(currentViewpointCamera)),
+                            Q_ARG(QVariant, hitResult[0]),
+                            Q_ARG(QVariant, hitResult[1]),
+                            Q_ARG(QVariant, hitResult[2]),
+                            Q_ARG(QVariant, hitResult[3]),
+                            Q_ARG(QVariant, hitResult[4]),
+                            Q_ARG(QVariant, hitResult[5]),
+                            Q_ARG(QVariant, hitResult[6]));
+  return location.value<QObject*>();
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::setTransformationMatrixInternal(
+    double quaternionX, double quaternionY, double quaternionZ, double quaternionW,
+    double translationX, double translationY, double translationZ)
+{
+  emit transformationMatrixChanged(quaternionX, quaternionY, quaternionZ, quaternionW,
+                                   translationX, translationY, translationZ);
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::setFieldOfViewInternal(double xFocalLength, double yFocalLength,
+                                             double xPrincipal, double yPrincipal,
+                                             double xImageSize, double yImageSize)
+{
+  emit fieldOfViewChanged(xFocalLength, yFocalLength,
+                          xPrincipal, yPrincipal,
+                          xImageSize, yImageSize);
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::renderFrameInternal()
+{
+  emit renderFrame();
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::setTranslationFactorInternal(double)
+{
+  // do nothing, translationFactorChanged signal is emitted in ArcGISArViewInterface::setTranslationFactor
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::setLocationInternal(double latitude, double longitude, double altitude)
+{
+  emit locationChanged(latitude, longitude, altitude);
+}
+
+/*!
+  \internal
+ */
+void QmlArcGISArView::setHeadingInternal(double heading)
+{
+  emit headingChanged(heading);
+}
+
+/*!
+  \internal
+
+  Resets the device tracking and related properties.
+ */
+void QmlArcGISArView::resetTrackingInternal()
+{
+  // to implmente
+}
+
+/*!
+  \internal
+
+  As the QML API headers is not available, it's not possible to use the concret types in the QmlArcGISArView
+  API. Then, this API uses QObject* and this function is used to verify than the type of the objects passed
+  in the parameters are correct (using the Qt's meta-object).
```

Comment:
```
\brief Verify than \a object is of the type expressed in \a className.
\internal

This method exists because the QML API headers are not available, so it's not possible to use the concrete types in the \c QmlArcGISArViewAPI.